refresh.template.py

```diff
@@ -212,36 +215,105 @@ def _get_cpp_command_for_files(compile_action: json):
         # This makes the output vastly smaller, which has been a problem for users.
-            # e.g. https://github.com/insufficiently-caffeinated/caffeine/pull/577 
+        # e.g. https://github.com/insufficiently-caffeinated/caffeine/pull/577
         # Without this, we emit an entry for each header for each time it is included, which is explosively duplicative--the same reason why C++ compilation is slow and the impetus for the new modules.
         # Revert when https://github.com/clangd/clangd/issues/123 is solved, which would remove the need to emit headers, because clangd would take on that work.
         # If https://github.com/clangd/clangd/issues/681, we'd probably want to find a way to filter to one entry per platform.
         header_files = [h for h in header_files if h not in header_file_entries_written]
         header_file_entries_written.update(header_files)
 
         for file in itertools.chain(source_files, header_files):
-            sys.stdout.write(json.dumps({
-                'file': file,
-                'command': command,
-                'directory': bazel_workspace_dir
-            }, indent=2, check_circular=False))
-            sys.stdout.write(',')
+            yield {
+                "file": file,
+                "command": command,
+                # Bazel gotcha warning: If you were tempted to use `bazel info
+                # execution_root` as the build working directory for
+                # compile_commands...search ImplementationReadme.md to learn why
+                # that breaks.
+                "directory": os.fspath(directory),
+            }
+
+
+# Call with the same flags as `bazel build`, one target per call, followed by flags
+# Yields the entries to compile_commands.json
+def get_commands(directory: pathlib.Path, target: str, flags: str) -> str:
+    # Log clear completion messages
+    print(f"\033[0;34m>>> Analyzing commands used in {target}\033[0m")
+
+    # Queries Bazel's C-family compile actions, and runs them through our extract.py reformatter
+    # Aquery docs if you need em: https://docs.bazel.build/versions/master/aquery.html
+    # One bummer, not described in the docs, is that aquery filters over *all* actions for a given target, rather than just those that would be run by a build to produce a given output. This mostly isn't a problem, but can sometimes surface extra, unnecessary, misconfigured actions. Chris has emailed the authors to discuss and filed an issue so anyone reading this could track it: https://github.com/bazelbuild/bazel/issues/14156.
+    # We switched to jsonproto instead of proto because of https://github.com/bazelbuild/bazel/issues/13404. We could change back when fixed--reverting most of the commit that added this line and tweaking the build file to depend on the target in that issue. That said, it's kinda nice to be free of the dependency, unless (OPTIMNOTE) jsonproto becomes a performance bottleneck compated to binary protos.
```

Comment:
Maybe let's toss this down next to the argument list itself? I only didn't do it before because of Bash's rigidities.

Reply:
Sorry, I am not sure what you mean. Split this up and distribute it into the `cmd` list?